Gems/ROS2/Code/CMakeLists.txt

```diff
@@ -59,6 +59,7 @@ ly_add_target(
             Source/VehicleDynamics
     BUILD_DEPENDENCIES
         PUBLIC
+            AZ::AssetBuilderSDK
```

Comment:
Nit: Could/should this be a private build dependency?

Reply:
This shouldn't be a dependency on the `ROS2.Static` target, as this should only be part of the Tool applications.

This should be added to the `ROS2.Editor.Static` target

Also this will result in a CMake configure error when host tools aren't available to be built, such as when using the Monolithic Build Permutation

Reply:
fixed! Private AssetBuilderSDK dependency and code and dependency are moved to Editor.Static now.